lib/posix/options/Kconfig.deprecated

```diff
@@ -5,4 +5,13 @@
 # This file should be removed after Zephyr 4.0 is released
 
 menu "Deprecated POSIX options"
+
+config POSIX_LIMITS_RTSIG_MAX
+	int "_POSIX_RTSIG_MAX value in limits.h [DEPRECATED]"
+	default 0
```

Comment:
Should we add `select DEPRECATED` here?

Reply:
Oddly, int options get a warning if they select deprecated, because they need to have a default value and are effectively always selected, AFAICT. I might be doing it wrong though. 

What I've done is add a build time `__WARN()` in select places if it is defined and non-zero.

Reply:
I guess it isn't as straightforward to deprecate a Kconfig of type `int`:
https://github.com/zephyrproject-rtos/zephyr/blob/4b6d3f1e16ea8ccccbf85f1eaf7efd0caea11766/kernel/Kconfig.smp#L41-L46

Reply:
Agreed. It seems best practice would be to default the deprecated value to the non-deprecated value - those should be fixed now.